src/plugins/src/volumes/grid_spherical.cpp

```diff
@@ -0,0 +1,90 @@
+#include <mitsuba/core/fresolver.h>
+#include <mitsuba/core/properties.h>
+#include <mitsuba/core/spectrum.h>
+#include <mitsuba/core/string.h>
+#include <mitsuba/core/transform.h>
+
+#include <mitsuba/render/srgb.h>
+#include <mitsuba/render/volume.h>
+#include <mitsuba/render/volumegrid.h>
+#include <enoki/dynamic.h>
+#include <enoki/tensor.h>
+
+NAMESPACE_BEGIN(mitsuba)
+
+enum class FilterType { Nearest, Trilinear };
+enum class WrapMode { Repeat, Mirror, Clamp };
+
+/**!
+.. _volume-gridvolume_spherical:
+
+Grid-based volume data source in spherical coordinates(:monosp:`gridvolume_spherical`)
+----------------------------------------------------------
+
+.. pluginparameters::
+
+ * - gridvolume
+   - |gridvolume|
+   - Nested gridvolume object that holds the actual data organized in spherical coordinates
+     Method calls are pre processed if necessary and then forwarded to this object.
+
+ * - inner_bsphere_radius
+   - |float|
+   - Specifies the relative radius of the inner bounding sphere of the spherical shell
+     that delimits this volume. This must be within [0, 1.0]. 
+
+This class implements a gridvolume in spherical coordinates. To achieve that, it holds a nested
+gridvolume object and preprocesses method calls accordingly before forwarding them to the nested
+object. In the :monosp:`eval()` method, the sampled point is converted to spherical coordinates,
+relative to the center of the sphere and converted such that :math:`r`, :math:`\theta` and :math:`\phi`
+cover the [0,1] range.
+
+*/
+
+template <typename Float, typename Spectrum>
+class GridVolumeSpherical final : public Volume<Float, Spectrum> {
+public:
+    MTS_IMPORT_BASE(Volume, m_to_local, update_bbox)
+    MTS_IMPORT_TYPES(Volume, VolumeGrid)
+
+    GridVolumeSpherical(const Properties &props) : Base(props) {
+        m_gridvol = props.volume<Volume>("gridvolume", 1.f);
+        m_inner_bsphere_radius_relative = props.get<ScalarFloat>("inner_bsphere_radius", 0.f);
+    }
```

Comment:
Constructor with full bounding box control:

```suggestion
    MTS_IMPORT_BASE(Volume, m_to_local)
    MTS_IMPORT_TYPES(Volume, VolumeGrid)

    GridVolumeSpherical(const Properties &props) {
        m_to_local = props.get<ScalarTransform4f>("to_world", ScalarTransform4f()).inverse();
        update_bbox();
        m_gridvol = props.volume<Volume>("gridvolume", 1.f);
        m_inner_bsphere_radius_relative = props.get<ScalarFloat>("inner_bsphere_radius", 0.f);
    }
```
Note that the call to the parent constructor is removed.

And further down, possibly as a public member:

```cpp
void update_bbox() {
    ScalarTransform4f to_world = m_to_local.inverse();
    ScalarPoint3f a = to_world * ScalarPoint3f(-1.f, -1.f, -1.f);
    ScalarPoint3f b = to_world * ScalarPoint3f(1.f, 1.f, 1.f);
    m_bbox = ScalarBoundingBox3f(a, b);
}